src/main/java/FiguresList.java

```diff
@@ -0,0 +1,7 @@
+public class FiguresList<K extends Figure> {
+    public K figure;
```

Comment:
what for do we need this class? probably it would be better to use ArrayList in your main and parametrized it with Figure(remember about PECS)